internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,508 @@
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for cluster"))
+			}
+
+			UID = iamResponse.Policy.Meta.UID
+		}
+	case workspaceScope:
+		if scopedFullname.fullnameWorkspace != nil {
+			iamRequest := &workspaceiammodel.VmwareTanzuManageV1alpha1WorkspacePatchWorkspaceIAMPolicyRequest{
+				FullName:         scopedFullname.fullnameWorkspace,
+				BindingDeltaList: blData,
+			}
+
+			iamResponse, err := config.TMCConnection.WorkspaceIAMResourceService.ManageV1alpha1WorkspaceIAMPolicyPatch(iamRequest)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for workspace"))
+			}
+
+			UID = iamResponse.Policy.Meta.UID
+		}
+	case namespaceScope:
+		if scopedFullname.fullnameNamespace != nil {
+			iamRequest := &namespaceiammodel.VmwareTanzuManageV1alpha1ClusterNamespacePatchNamespaceIAMPolicyRequest{
+				FullName:         scopedFullname.fullnameNamespace,
+				BindingDeltaList: blData,
+			}
+
+			iamResponse, err := config.TMCConnection.NamespaceIAMResourceService.ManageV1alpha1ClusterNamespaceIAMPolicyPatch(iamRequest)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "unable to create Role Binding for namespace"))
+			}
+
+			UID = iamResponse.Policy.Meta.UID
+		}
+	case unknownScope:
+		log.Fatalf("[ERROR]: No valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scopesAllowed[:], `, `))
+	}
+
+	d.SetId(UID)
+
+	return append(
+		diags,
+		resourceIAMPolicyRead(context.WithValue(ctx, contextMethodKey{}, createKey), d, m)...,
+	)
+}
+
+// nolint: funlen, gocognit, gocyclo
+// resourceIAMPolicyRead returns the intersection between binding list in terraform state and TMC server.
+func resourceIAMPolicyRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	var (
+		meta         *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta
+		policyList   []*iammodel.VmwareTanzuCoreV1alpha1PolicyIAMPolicy
+		rbServerList []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding
+		calRBList    []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding
+	)
+
+	scopedFullnameData := constructScope(d)
+	rbStateList := constructRoleBindingList(d)
+
+	if scopedFullnameData == nil {
+		return diags
+	}
+
+	switch scopedFullnameData.scope {
+	case organizationScope:
+		if scopedFullnameData.fullnameOrganization != nil {
+			resp, err := config.TMCConnection.OrganizationIAMResourceService.ManageV1alpha1OrganizationIAMPolicyGet(scopedFullnameData.fullnameOrganization)
+			if err != nil || resp == nil {
+				if clienterrors.IsNotFoundError(err) {
+					d.SetId("")
+					return diags
+				}
+
+				return diag.FromErr(errors.Wrapf(err, "unable to get Role Bindings for organization"))
+			}
+
+			fullName := flattenScope(scopedFullnameData)
```

Comment:
Here we have to use the response data.

Reply:
@Vasundharashukla Refer to this:

https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/009f44908c1424fe9df50b31b95d1e44d5bdab1f/internal/resources/policy/type/security/resource_security_policy.go#L148